From f77c5f479d54e6a52cf101d9435a188cb5e3d3b9 Mon Sep 17 00:00:00 2001
From: Ted John <ted@brambles.org>
Date: Thu, 17 Dec 2020 19:01:30 +0000
Subject: [PATCH] Patch duktape

---
 ports/duktape/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ports/duktape/CMakeLists.txt b/ports/duktape/CMakeLists.txt
index 746aaa910..19e6af1dd 100644
--- a/ports/duktape/CMakeLists.txt
+++ b/ports/duktape/CMakeLists.txt
@@ -20,6 +20,9 @@ set_target_properties(duktape PROPERTIES PUBLIC_HEADER "${DUKTAPE_HEADERS}")
 set_target_properties(duktape PROPERTIES VERSION ${duktape_VERSION})
 set_target_properties(duktape PROPERTIES SOVERSION ${duktape_MAJOR_VERSION})
 
+target_compile_definitions(duktape PRIVATE -DDUK_USE_CPP_EXCEPTIONS)
+target_compile_definitions(duktape PRIVATE -DDUK_USE_DATE_NOW_WINDOWS)
+
 if (BUILD_SHARED_LIBS)
   target_compile_definitions(duktape PRIVATE -DDUK_F_DLL_BUILD)
 endif ()
-- 
2.29.2.windows.3

